### PR TITLE
Unit List: Sort by XP as second criterion when sorting by level.

### LINKED ISSUE #3260: Unit List sorting by level doesn't sort by XP as second criterion (regression from 1.12)
In the "Unit list" dialog (Alt+u), when sorting by unit level (highest first), in 1.12 units of the same level were sorted so that units with more XP came first. In 1.14 that no longer happens: if I have one unit 1XP from advancing and other units at 0XP, the 1XP-to-advance unit is listed in the middle (in the same position it took in the unsorted list).

This was implemented for the recall dialog in #1738.

### PR DESCRIPTION
**Pull request by jostephd**
Fixes #3260.

### CODE REVIEW

File: src/gui/dialogs/unit_list.cpp [hunk 1]
```diff
@@ -164,7 +165,10 @@ void unit_list::pre_show(window& window)
 	list.register_translatable_sorting_option(1, [this](const int i) { return unit_list_[i]->name().str(); });
 	list.register_sorting_option(2, [this](const int i) { return unit_list_[i]->movement_left(); });
 	list.register_sorting_option(3, [this](const int i) { return unit_list_[i]->hitpoints(); });
-	list.register_sorting_option(4, [this](const int i) { return unit_list_[i]->level(); });
+	list.set_column_order(4, {{
+			[this](int lhs, int rhs) { return unit_recall::unit_recall_default_compare(unit_list_[rhs], unit_list_[lhs]); },
```
Comment (Vultraz) [inline src/gui/dialogs/unit_list.cpp]: Please fix the indent.

File: src/gui/dialogs/unit_recall.hpp [hunk 2]
```diff
@@ -45,6 +45,9 @@ class unit_recall : public modal_dialog
 		return selected_index_;
 	}
 
+	/** Function to sort recall_list_ by default. */
+	static bool unit_recall_default_compare(const unit_const_ptr first, const unit_const_ptr second);
```
Comment (Vultraz) [inline src/gui/dialogs/unit_recall.hpp]: Could you rename this to just `default_compare`.

**Comment by jostephd** [comment]
Done and done. Shall I squash the new commits?

**Comment by jostephd** [comment]
Here you go.

**Pull request merged**